test/e2e/framework.go

```diff
@@ -896,7 +930,24 @@ func (data *TestData) deployFlowAggregator(ipfixCollector string) error {
 	return nil
 }
 
-func (data *TestData) mutateFlowAggregatorConfigMap(ipfixCollectorAddr string) error {
+// deployIngressClickHouse deploys ingress-nginx-controller
+func (data *TestData) deployIngressNginxController() error {
+	rc, _, _, err := data.provider.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl apply -f %s", ingressNginxControllerYML))
+	if err != nil || rc != 0 {
+		return fmt.Errorf("error when deploying the ingress-nginx-controller; %s not available on the control-plane Node", ingressNginxControllerYML)
+	}
+	// Check for ingress-nginx-controller Pod running again for ingress creation
+	ingressPod, err := data.getIngressNginxController()
+	if err != nil {
+		return fmt.Errorf("error when getting ingress-nginx-controller Pod: %v", err)
+	}
+	if err = data.podWaitForReady(2*defaultTimeout, ingressPod.Name, ingressNginxNamespace); err != nil {
```

Comment:
there has to be a better way to check for the nginx controller to be ready. Could you check the status of the Deployment instead?

Reply:
Yes. I've updated it to check the status of the Deployment instead.